tests/helpers_tests/test_condition.py

```diff
@@ -17,117 +17,342 @@
+
+    condition_list = [['weird_condition', 'hi', 'weird_type', 'exact']]
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      condition_list, {'weird_condition': 'hi'}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exists__returns_false__when_no_user_provided_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exists__returns_false__when_user_provided_value_is_null(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': None}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_string(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 'hi'}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_number(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 10}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': 10.0}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exists__returns_true__when_user_provided_value_is_boolean(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exists_condition_list, {'input_value': False}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exact_string__returns_true__when_user_provided_value_is_equal_to_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': 'Lacerta'}
+    )
+
+    self.assertStrictTrue(evaluator.evaluate(0))
+
+  def test_exact_string__returns_false__when_user_provided_value_is_not_equal_to_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': 'The Big Dipper'}
+    )
+
+    self.assertStrictFalse(evaluator.evaluate(0))
+
+  def test_exact_string__returns_null__when_user_provided_value_is_different_type_from_condition_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {'favorite_constellation': False}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exact_string__returns_null__when_no_user_provided_value(self):
+
+    evaluator = condition_helper.CustomAttributeConditionEvaluator(
+      exact_string_condition_list, {}
+    )
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+  def test_exact_number__returns_true__when_user_provided_value_is_equal_to_condition_value(self):
```

Comment:
We should have two versions of this:
* `test_exact_int_returns_true__when_user_provided_value_is_equal_to_condition_value`
* `test_exact_float_returns_true__when_user_provided_value_is_equal_to_condition_value`
with the same assertions but different condition value types.

Same for most or all of the other `exact_number`, `greater_than`, and `less_than` tests!